### ISSUE TITLE
Pick the right observatory when multiple are available

### ISSUE THREAD

**Issue by xster** [Dart-Code/Dart-Code#2722]
Sometimes when I attach to the selected device, I get 

![Screen Shot 2020-08-14 at 4 50 46 PM](https://user-images.githubusercontent.com/156888/90300219-4da9a400-de4e-11ea-9403-db0baf3ff5da.png)

But theoretically, VSCode should know what the currently open project's app id should be and should just send it along.

**Comment by DanTup** [comment]
How do you get into the state where this happens?

Where do these IDs come from? Are they the same for each version of the app, or different for each platform?

VS Code doesn't currently know anything about the file structure of the native apps, so if they're in there I don't think we should go fishing for them. Maybe it would be better if there was a JSON version of this error (in `--machine` mode) that lets the editor present the options to the user and have them click them.

@jonahwilliams what do you think?

We already have some daemon-to-client requests (for `expose-url`) so we could potentially do something like that too (rather than terminating, just ask the client to make a choice and then continue when it responds).

**Comment by xster** [comment]
I think it was just from running one Flutter app, not closing the app (such as launching from CLI then detach). Repeat for a second app. Then go to VSCode for one of the projects and attach to Flutter on device.

**Comment by DanTup** [comment]
Perhaps another simpler option here would be for `flutter attach` to automatically use the "current app" if it's been executed from within a project folder (or have a flag to do so)? That way the editor still doesn't need to know any of the platform-specific stuff, and Flutter presumably has easy access and knowledge to get the ID for the project?

**Comment by jonahwilliams** [comment]
The tool could try and determine whether their is a usable identifier in the current project. 

If a terminal attached, we could prompt with the default being the detected project if found. If there is no terminal we can select the detected project, or exit if no project was detected.

WDUT?

**Comment by DanTup** [comment]
I don't think this is an issue any more, because I think it automatically connects to the last VM Service from the log. If this isn't the case and you know how to repro, please let me know!
